### PR TITLE
Archive, check,json and exec command with firejail for inspec

### PR DESCRIPTION
**Pull request by YashviJain01**
### :nut_and_bolt: Description: What code changed, and why?

Inspec archive, json and check commands will be run in the sandbox environment now.

For executing the command in sandbox env, the file is copyied into newly created tmp directory in /tmp folder.

We are reading the output using standard output file and standard error files.

### :chains: Related Resources

https://chefio.atlassian.net/browse/CHEF-5691

### :+1: Definition of Done

A sandbox environment is created with a profile now.

### :athletic_shoe: How to Build and Test the Change

```
rebuild components/compliance-service/
```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [X] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [X] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [X] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [X] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [X] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [X] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [X] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [X] Tests added/updated? (all new code needs new tests)
- [X] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/compliance-service/inspec/cli.go [hunk 1]
```diff
@@ -78,22 +94,41 @@ func Scan(paths []string, target *TargetConfig, timeout time.Duration, env map[s
...
+		return nil, nil, NewInspecError(INVALID_PARAM, err.Error())
+	}
+
+	tmpKeys := copyKeyFilesIntoTmpDirectory(tmpDirPath, target.KeyFiles)
+	newTarget := target
+	newTarget.KeyFiles = tmpKeys
+	stdoutFile, erroutFile, shellFile := shellscriptAndResponse(exec_command, tmpDirPath)
+	commandArgs := args
+	commandArgs = append(commandArgs, []string{"/bin/sh", shellFile}...)
+	commandArgs = append(commandArgs, paths...)
```
Comment (Sahiba3108) [inline components/compliance-service/inspec/cli.go:114]: Can we combine line no 112 to 114 in 1 line.

File: components/compliance-service/inspec/cli.go [hunk 2]
```diff
@@ -78,22 +94,41 @@ func Scan(paths []string, target *TargetConfig, timeout time.Duration, env map[s
...
+			return nil, nil, NewInspecError(UNKNOWN_ERROR, errString)
 		}
 		args = append(args, fmt.Sprintf("--input-file %s", filename))
 	}
+	//Changing the home directory to tmp directory created
+	env["HOME"] = tmpDirPath
 
-	stdOut, stdErr, err := run(args, target, timeout, env)
+	logrus.Debugf("Run: inspec %v", args)
+	_, _, err = run(commandArgs, newTarget, timeout, env)
```
Comment (Sahiba3108) [inline components/compliance-service/inspec/cli.go:128]: We are not checking this err.

File: components/compliance-service/inspec/cli.go [hunk 3]
```diff
@@ -264,33 +320,94 @@ func Check(profilePath string) (CheckResult, error) {
...
+	logrus.Infof("Successfully checked inspec profile in %s", profilePath)
 	return res, nil
 }
 
-func Json(profilePath string) ([]byte, error) {
-	args := []string{shimBinName, "json", profilePath}
-	logrus.Debugf("Run: inspec %v", args)
-	stdout, stderr, err := run(args, nil, defaultTimeout, inspecShimEnv())
-	logrus.Debugf("Run: %s %s %v", stdout, stderr, err)
+func Json(profilePath string, firejailprofilePath string) ([]byte, error) {
```
Comment (Sahiba3108) [inline components/compliance-service/inspec/cli.go:327]: can we combine the logic of Json and Check function.

**Review by Sahiba3108 — CHANGES_REQUESTED** [review]

File: components/compliance-service/inspec/cli.go (hunk 2 above)
Comment (YashviJain01) [reply components/compliance-service/inspec/cli.go:128]: this error we are not supposed to check, as we are checking the sucess and json files.

File: components/compliance-service/inspec/cli.go (hunk 3 above)
Comment (YashviJain01) [reply components/compliance-service/inspec/cli.go:327]: Nopes, these are two commands which needed to be run parlely. Both doing the different operations. This was like this before, we just added few things

File: components/compliance-service/inspec/cli.go (hunk 1 above)
Comment (YashviJain01) [reply components/compliance-service/inspec/cli.go:114]: We can, but both of the lines are doing the same thing. It's just making the code in one line. This is making it more readable. @kalroy thoughts?

**Review by kalroy — APPROVED** [review]

**Review by Sahiba3108 — APPROVED** [review]

**Review by daveaugustus — APPROVED** [review]

**Pull request merged**